FreeRTOS/FreeRTOS-Plus-TCP
# IPv6/Multi: Reduced complexity FreeRTOS_TCP_IP

Description
-----------
The module FreeRTOS_TCP_IP.c had three "complex" functions, which will be split-up in smaller parts by adding helper functions:

prvTCPReturnPacket()  score **77**
- prvTCPReturn_SwapAddresses()
    - prvTCPReturn_SetChecksum()
- prvTCPReturn_SetEndPoint()
- prvTCPReturn_CheckTCPWindow()
- prvTCPReturn_SetSequenceNumber()

prvTCPStatusAgeCheck() score 5
- prvTCPStatusNeedsCheck()

prvTCPPrepareSend()  score **47**
- prvTCPPrepare_HandleClose()
- prvTCPPrepare_KeepAlive()

xProcessReceivedTCPPacket()  score **58**
- prvProcessReceived_RSTPacket()


Test Steps
-----------
The changes have been tested in real-life operations.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
